src/field_theory/perfect_closure.lean

```diff
@@ -426,3 +426,11 @@ end
 end field
 
 end perfect_closure
+
+/-- A field with prime characteristic and surjective frobenius map is perfect. -/
+noncomputable def perfect_ring.of_surjective (k : Type*) [field k] (p : ℕ) [fact p.prime]
```

Comment:
Is the field assumption on `k` necessary? I'm fairly sure you could get by with much weaker assumptions on `k`.

Reply:
`[comm_ring k] [is_domain k]` is surely enough, we have [function.injective](https://leanprover-community.github.io/mathlib_docs/init/function.html#function.injective) [⇑](https://leanprover-community.github.io/mathlib_docs/init/coe.html#coe_fn)([frobenius](https://leanprover-community.github.io/mathlib_docs/algebra/char_p/basic.html#frobenius) R p).

Reply:
Ah, nice!